src/pybullet_helpers/robots/single_arm.py

```diff
@@ -115,259 +192,190 @@ def closed_fingers(self) -> float:
+        assert len(joint_positions) == len(self.arm_joints)
+        for joint_id, joint_val in zip(self.arm_joints, joint_positions):
+            p.resetJointState(
+                self.robot_id,
+                joint_id,
+                targetValue=joint_val,
+                targetVelocity=0,
+                physicsClientId=self.physics_client_id,
+            )
+
+    def set_motors(self, joint_positions: JointPositions) -> None:
+        """Update the motors to move toward the given joint positions."""
+        assert len(joint_positions) == len(self.arm_joints)
 
-    @abc.abstractmethod
-    def set_motors(self, joints_state: JointsState) -> None:
-        """Update the motors to move toward the given joints state."""
-        raise NotImplementedError("Override me!")
+        # Set arm joint motors.
+        if CFG.pybullet_control_mode == "position":
+            p.setJointMotorControlArray(
+                bodyUniqueId=self.robot_id,
+                jointIndices=self.arm_joints,
+                controlMode=p.POSITION_CONTROL,
+                targetPositions=joint_positions,
+                physicsClientId=self.physics_client_id,
+            )
+        elif CFG.pybullet_control_mode == "reset":
+            self.set_joints(joint_positions)
+        else:
+            raise NotImplementedError("Unrecognized pybullet_control_mode: "
+                                      f"{CFG.pybullet_control_mode}")
 
-    @abc.abstractmethod
-    def forward_kinematics(self, joints_state: JointsState) -> Pose3D:
+    def go_home(self) -> None:
+        """Move the robot to its home end-effector pose."""
+        self.set_motors(self.initial_joint_positions)
+
+    def forward_kinematics(self, joint_positions: JointPositions) -> Pose3D:
         """Compute the end effector position that would result if the robot arm
-        joints state was equal to the input joints_state.
+        joint positions was equal to the input joint_positions.
 
         WARNING: This method will make use of resetJointState(), and so it
         should NOT be used during simulation.
         """
-        raise NotImplementedError("Override me!")
-
-    @abc.abstractmethod
-    def inverse_kinematics(self, end_effector_pose: Pose3D,
-                           validate: bool) -> JointsState:
-        """Compute a joints state from a target end effector position.
-
-        The target orientation is always self._ee_orientation.
+        self.set_joints(joint_positions)
+        ee_link_state = get_link_state(
+            self.robot_id,
+            self.end_effector_id,
+            physics_client_id=self.physics_client_id)
+        position = ee_link_state.worldLinkFramePosition
+        return position
 
-        If validate is True, guarantee that the returned joints state
-        would result in end_effector_pose if run through
-        forward_kinematics.
+    def _validate_joint_positions(self, joint_positions: JointPositions,
```

Comment:
It looks like this method isn't used anywhere right now -- should we just remove it until we need it (if ever)?

Reply:
It'll come handy once we integrate IKFast as we need to validate the IK solutions.

Reply:
makes sense -- can we remove it for the sake of merging this PR then, since it will probably be easier to test in a future PR where we add IKFast?

Reply:
Sure - will do that